bindings.c

```diff
@@ -662,6 +662,42 @@ static char *slurp_file(const char *from, int fd)
 	return contents;
 }
 
+
+/*
+ * Determine if 2 processes are in the same type of namespace
+ * When two processes are in the same ns, return >0, not same return 0, error returns <0
+ */
+static int in_same_ns(pid_t pid1, pid_t pid2, char *ns_type)
+{
+    char ns_path1[1024], ns_path2[1024];
+    int s1, s2;
+    struct stat buf1, buf2;
+
+    s1 = snprintf(ns_path1, sizeof(ns_path1)-1, "/proc/%d/ns/%s", pid1, ns_type);
+    if (s1 > 0)
+        ns_path1[s1] = '\0';
```

Comment:
Hm, that's not really correct as you're only checking the inode numbers but not the device numbers.
I've added functions in LXC that deal with this correctly. Just copy them over, please:
https://github.com/lxc/lxc/blob/faec88e5ac556d41fb4ee3c12f6fe956070ed516/src/lxc/attach.c#L158
https://github.com/lxc/lxc/blob/345a21ca9ec1b736208611f4bec7e24097ce279b/src/lxc/utils.c#L1379

They may contain a few macros that you might have to get rid off or copy over too and you might need to adapt them a little but I think they are better templates.
Also, if you only use this function to answer 'yes' or 'no', please return a bool.